docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,220 @@
+
+# Summary
+
+We need a new api that will allow us to skip cluster/host validations, this will be for unsupported use.
+This API should work on a per cluster basis and will have it's own feature flag to allow us to enable it for specific users.
+An event should be emitted whenever the API is used during an installation.
+Clusters containing skipped validations should not contribute to statistics as these may be skewed.
+# Motivation
+
+Some customer and partners have very specific HW that doesn't pass our validations and we want to allow them to proceed with installation.
+Sometimes we have bugs in our validations that block people from installing and we don't want our partners left in a position where they can't install.
+
+### Goals
+
+- Allow validations to be bypassed so that installation may proceed.
+
+### Non-Goals
+
+- None identified at this stage
+
+## Proposal
+
+There are three main components as I see it
+
+### The REST API
+
+#### Endpoint: List existing host validations.
+
+GET api/v2/cluster/<id>/skipped-host-validations/
+Get a list of all skipped host validations.
+```
+{
+    "no-ip-collisions-in-network",
+    "sufficient-network-latency-requirement-for-role"
+  ]
+}
+```
+#### Endpoint - List existing cluster validations.
+
+GET api/v2/cluster/<id>/skipped-cluster-validations/
+Get a list of all skipped cluster validations.
+```
+{
+    "network-prefix-valid",
+    "api-vips-defined"
+}
+```
+
+#### Endpoint - Add a skipped validation.
+
+PUT api/v2/cluster/<id>/skipped-host-validations/no-ip-collisions-in-network
+PUT api/v2/cluster/<id>/skipped-cluster-validations/api-vips-defined
+
+* Will add an entry to the list of validations to be skipped. This will appropriately de-deuplicated and no error will be returned if duplication is attempted.
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+#### Endpoint - Delete a skipped host validation.
+DELETE api/v2/cluster/<id>/skipped-host-validations/no-ip-collisions-in-network
+
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+#### Endpoint - Delete a skipped cluster validation.
+DELETE api/v2/cluster/<id>/skipped-cluster-validations/no-ip-collisions-in-network
+
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+#### Endpoint - Delete all skipped validations for host.
+DELETE api/v2/cluster/<id>/skipped-host-validations
+
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+#### Endpoint - Delete all skipped validations for cluster.
+DELETE api/v2/cluster/<id>/skipped-cluster-validations
+
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+
+### The Kubernetes API
```

Comment:
Again, seems overly complex - we can just add an annotation to skip all validations

Reply:
I don't think that we need something like that for kube-api in the first place. we have the unsupported way to disable validation on service level, i don't see a reason to have it in a cluster level. 
I don't think that ACM will have multiple different clients that will need to skip validations for a specific cluster.
At least this is my take on that.